plugins/knife/_knife

```diff
@@ -198,4 +227,16 @@ _cookbook_versions() {
   (knife cookbook show $words[4] | grep -v $words[4] | grep -v -E '\]|\[|\{|\}' | sed 's/ //g' | sed 's/"//g')
 }
 
+# Searches up from current directory to find the closest folder that has a .chef folder 
+# Useful for the knife upload/from file commands 
+_chef_root () {
+  slashes=${PWD//[^\/]/}
+  directory="$PWD"
+  for (( n=${#slashes}; n>0; --n ))
+  do
+    test -e "$directory/.chef" && echo "$directory" && return
+    directory="$directory/.."
```

Comment:
You can use `${directory:h}` to get the parent folder. Then you only need to check if `$directory = /` in your loop condition, and you don't need the slashes and the `for`.